images/origin/scripts/recycler.sh

```diff
@@ -4,40 +4,28 @@
 # this script is intended to be used in a pod that performs the scrub.
 # the container in the pod should succeed or fail based on the exit status of this script.
 
-set -o errexit
 set -o nounset
 set -o pipefail
 
-function recycle(){
+# first and only arg is the directory to scrub
+dir=${1:-}
 
-    # first and only arg is the directory to scrub
-    dir=$1
-
-    if [ -z "$dir" ]; then
-        echo "Usage:  scrub_directory some/path/to/scrub"
-        return 1
-    fi
-
-    if [ ! -e $dir ]; then
-        echo "scrub directory $dir does not exist"
-        return 1
-    fi
+if [ -z $dir ]; then
+    echo "Usage:  recycler.sh some/path/to/scrub"
+    exit 1
+fi
 
-    # remove everything but keep the directory for re-use as a volume
-    rm -rfv $dir/*
+echo "Scrubbing $dir"
 
-    if [ -z "$(ls -A $dir)" ]; then
-        echo "scrub directory $dir is empty"
-        return 0
-    else
-        echo "scrub directory $dir is not empty"
-        return 1
-    fi
-}
+rm -rf $dir/*
```

Comment:
"${dir}/*" does not work because rm does not support wildcard expansion.   This change wrongly returned exit code 0 and the script thought itself a success.